File: python/meteroid/models/product_event_data.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .product_family_id import ProductFamilyId
from .product_fee_type_enum import ProductFeeTypeEnum
from .product_id import ProductId


@dataclasses.dataclass
class ProductEventData(BaseModel):
    created_at: datetime

    fee_type: ProductFeeTypeEnum

    name: str

    product_family_id: ProductFamilyId

    product_id: ProductId

    description: t.Optional[str] = None

File: go/product_event_data.go
// this file is @generated
package meteroid

import "time"

type ProductEventData struct {
	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Description *string `json:"description,omitempty"`

	FeeType ProductFeeTypeEnum `json:"fee_type"`

	Name string `json:"name"`

	ProductFamilyId ProductFamilyId `json:"product_family_id"`

	ProductId ProductId `json:"product_id"`
}
